sklearn/metrics/cluster/unsupervised.py

```diff
@@ -129,6 +141,11 @@ def silhouette_samples(X, labels, metric='euclidean', **kwds):
         allowed by :func:`sklearn.metrics.pairwise.pairwise_distances`. If X is
         the distance array itself, use "precomputed" as the metric.
 
+    sample_weight : array, shape = [n_samples], optional
```

Comment:
I don't have a reference for it. That `sample_weight` in clustering should be treated as if you've compressed duplicates into weighted single-instances forces this interpretation, and it's how I implemented it in DBSCAN.